controllers/agents/app/serviceclaim_controller.go

```diff
@@ -178,12 +190,76 @@ func (r *ServiceClaimReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	return ctrl.Result{}, nil
 }
 
+func (r *ServiceClaimReconciler) ensureServiceClaimIsValid(
+	ctx context.Context,
+	sclaim *primazaiov1alpha1.ServiceClaim,
+) (bool, error) {
+	// check immutable properties
+	pp := r.getUpdatedImmutableProperties(ctx, sclaim)
```

Comment:
OK so basically you compare with original values and if there is a change fail and don't allow the change. Have you tried to play around with this https://kubernetes.io/blog/2022/09/29/enforce-immutability-using-cel/#append-only-list-of-containers. For example for append only the rule is this rule="oldSelf.all(x, x in self)" what it you had a rule like this rule="oldSelf.all(x, x in self) && self.all(x, x in oldSelf)" would that work?

Reply:
They would be very nice to use, but actually they're [beta in k8s v1.25](https://kubernetes.io/blog/2022/09/23/crd-validation-rules-beta/). We may face problems with older k8s versions

Reply:
As decided use CEL